_posts/2016/02/2016-02-03-swdcretreat2015-university_course.html

```diff
@@ -0,0 +1,437 @@
+  </ul>
+  I try to give a big picture in the beginning of class (e.g. websites hosted on github)
+  then show them organized code repository (e.g. a formated readme and folder structure).
+  However, I do not get to visualization until near the end of 4th or 5th day.
+</p>
+
+<p>
+  <h4>
+    The Gapminder dataset is on CRAN for the R folks, what about the Python people? Is there a CSV?
+  </h4>
+</p>
+
+<p>
+  [J] The <a href='https://cran.r-project.org/web/packages/gapminder/index.html'>package</a>
+  itself contains the <a href='https://github.com/jennybc/gapminder/tree/master/inst'>TSV</a> in non https locations, and <a href='https://github.com/jennybc/gapminder'>github</a>.
+</p>
+
+<p>
+  [T] Nancy Soontiens from UBC has developed a <a href='http://nsoontie.github.io/2015-03-05-ubc/novice/python/Pandas-Lesson.html'>gapminder lesson for Python</a> for the women in science workshop
+</p>
+
+<p>
+  <h4>
+    Jenny teaches a 13 week course and Tiffany you teach a 3.5 week course, How much time per week do you meet? And how do you split up the SWC material thoughout the weeks/days
+  </h4>
+</p>
+
+<p>
+  [J] I teach less classic SWC.  I have 3 hours with the students a week.
+  About 80% of the time is setting the stage with slides.
+  The rest is working in R.
+  The way I do it, most of the learning is in the homeworks.
+  The class is just enough information and guidance so they are not totally lost during the homeworks.
+</p>
+
+<p>
+  [T] We do 3 hours of shell the first day.
+  3 Hours of git the next.
+  I think it depends how many active learning exercise you can get.
+  For me it took me 9 Hours to get through loops, functions, and if statements because I had a lot of active learning exercise because I wanted them to fully understand these concepts.
+  Testing took another 3 hours (in the third week of the course).
+  Essentially I do 2.5 weeks of SWC then we do code review.
+  Each student is given the other student's homework to review.  And I sit with them and they give me a summary of the pull request.
+  At the same time the rest of the class are working on their final project.
+</p>
+
+<!--
+[D] What are your opintions on a half-semester course.  Or do you think it's best to be a
+-->
+
+<p>
+  <h4>
+    What final projects do you have the students work on?  Do you assign them a project or do they bring their own questions?
+  </h4>
+</p>
+
+<p>
+  [T] I've been extremely flexible with this.
+  I have them think about the final project from Day 1.
+  They pick something in the area they are interested in and have a research question with a hypothesis and do something that requires a computer to do.
+  These are undergraduates, so my expectations are not as high because they do not have as much research experience.
+  They must have a Python script import data and save a plot or table from an analysis or visualization
+  and have a record in either a Makefile or Shell script to make sure everything is reproducible.
```

Comment:
Please change "have a record in" to "have a record of the Python script inputs in"